app/models/user_block.rb

```diff
@@ -27,6 +27,12 @@ def active?
   end
 
   ##
+  # returns true if the block is a "zero hour" block
+  def zero_hour?
+    needs_view && (ends_at.to_i == updated_at.to_i)
```

Comment:
Looking at the DB entries the times differ in the nano seconds, AFAIK to_i returns seconds since the epoch, just a ham fisted way of reducing precision.